### PR TITLE
CP-40716: update usages of third-party libraries

### PR DESCRIPTION
**Pull request by psafont**
Adapts code to new versions introduced by https://github.com/xapi-project/xs-opam/pull/610

fixes https://github.com/xapi-project/xen-api/issues/4686

### CODE REVIEW

File: ocaml/gencert/selfcert.ml [hunk 1]
```diff
@@ -120,16 +120,17 @@ let generate_pub_priv_key length =
     |> X509.Private_key.decode_pem
     |> R.reword_error (fun _ -> R.msg "decoding private key failed")
   in
+  let err_not_rsa = R.error_msg "generated private key does not use RSA" in
   let* rsa =
-    try match privkey with `RSA x -> Ok x
-    with _ -> R.error_msg "generated private key does not use RSA"
+    try match privkey with `RSA x -> Ok x | _ -> err_not_rsa
+    with _ -> err_not_rsa
```
Comment (edwintorok) [inline ocaml/gencert/selfcert.ml:126]: Is the `try` really needed here? If you handle the wildcard in the inner match then the match itself cannot raise exceptions, and there is nothing else there to raise exceptions.

**Review by edwintorok — APPROVED** [review]
LGTM they are mostly name changes to use the new APIs.
One minor comment about the use of try+match+with+with (although that wasn't introduced by this PR).

Also not sure about the lazy default context, but as long as the tests pass I think it'll be fine.

**Comment by psafont** [comment]
> Also not sure about the lazy default context, but as long as the tests pass I think it'll be fine.
See https://github.com/mirage/ocaml-conduit/pull/395

**Review by robhoes — APPROVED** [review]

**Review by lindig — APPROVED** [review]

File: ocaml/message-switch/async/protocol_async.ml [hunk 2]
```diff
@@ -56,7 +56,7 @@ module M = struct
       Monitor.try_with ~extract_exn:true connect >>= function
       | Error
           (Unix.Unix_error
-            (Core.(Unix.ECONNREFUSED | Unix.ECONNABORTED | Unix.ENOENT), _, _)
+            (Core_unix.(ECONNREFUSED | ECONNABORTED | ENOENT), _, _)
```
[the remark this answers is not in the document]
Comment (psafont) [reply ocaml/message-switch/async/protocol_async.ml:59]: Note the trailing `, _, _`

**Pull request merged**